code/modules/cards/cards.dm

```diff
@@ -0,0 +1,128 @@
+/*
+** The base card class that is used by decks and cardhands
+*/
+/obj/item/toy/cards
+	resistance_flags = FLAMMABLE
+	max_integrity = 50
+	/// Do all the cards drop to the floor when thrown at a person
+	var/can_play_52_card_pickup = TRUE
+	/// List of cards for a hand or deck
+	var/list/cards = list()
+
+/obj/item/toy/cards/Destroy()
+	if(LAZYLEN(cards))
+		QDEL_LIST(cards)
+	return ..()
+
+/// This is how we play 52 card pickup
+/obj/item/toy/cards/throw_impact(mob/living/target, datum/thrownthing/throwingdatum)
+	if(..() || !istype(target)) // was it caught or is the target not a living mob
+		return
+
+	if(!throwingdatum?.thrower) // if a mob didn't throw it (need two people to play 52 pickup)
+		return
+
+	var/mob/living/thrower = throwingdatum.thrower
+
+	var/has_no_cards = !LAZYLEN(cards)
+	if(has_no_cards)
+		return
+
+	for(var/obj/item/toy/singlecard/card in cards)
+		cards -= card
+		card.forceMove(target.drop_location())
+		if(prob(50))
+			card.Flip()
+		card.pixel_x = rand(-16, 16)
+		card.pixel_y = rand(-16, 16)
+		var/matrix/M = matrix()
+		var/angle = pick(0, 90, 180, 270)
+		M.Turn(angle)
+		card.transform = M
+		card.update_appearance()
+
+	playsound(src, 'sound/items/cardshuffle.ogg', 50, TRUE)
+
+	if(istype(src, /obj/item/toy/cards/deck))
```

Comment:
istype(src)'s are very often a bad pattern, why not just add this section to /obj/item/toy/cards/deck/throw_impact() ?

Reply:
Agreed.  There were a lot of istype's in the old card code that I deleted.  Fixed with https://github.com/tgstation/tgstation/pull/64200/commits/a5e5cbe0df8644a2185c96a2c0ce9fe04f21b7b0